cmake/Dependencies.cmake

```diff
@@ -126,9 +128,14 @@ if (BUILD_PROTOBUF)
   endif()
 
   include_directories(SYSTEM ${Protobuf_INCLUDE_DIRS})
-  list(APPEND DALI_LIBS ${Protobuf_LIBRARY})
+  list(APPEND DALI_LIBS protobuf::libprotobuf)
   # hide things from the protobuf, all we export is only is API generated from our proto files
   list(APPEND DALI_EXCLUDES libprotobuf.a)
+  # find all the libraries that protobuf::libprotobuf depends on
+  get_link_libraries(PROTO_LIB_DEPS protobuf::libprotobuf)
```

Comment:
So adding the `protobuf::libprotobuf` for DALI_DEPS is enough to build against protobuf; finding the transitive closure is only for the EXCLUDES?

Reply:
Because `DALI_LIBS` is passed to `target_link_libraries` which can deal with extending this to the right values, while `DALI_EXCLUDES` is used literally as a part of the linker custom command
```target_link_libraries(dali PRIVATE "-Wl,--exclude-libs,${exclude_libs}")``` where we need particular values after `-Wl,--exclude-libs,` so CMake just appends sting here while in the former call, it does all the magic for us.

Reply:
Thanks!